File: dart-wrapper/cardinal_sdk/ios/Classes/CardinalSdkPlugin.swift
import Flutter
import CardinalDartSdkSupportLib

public class CardinalSdkPlugin: NSObject, FlutterPlugin {
	public static func register(with registrar: FlutterPluginRegistrar) {
		ApiScope.shared.setup()
		let apiChannel = FlutterMethodChannel(name: "com.icure.cardinal.sdk/api", binaryMessenger: registrar.messenger())
		let initializersChannel = FlutterMethodChannel(name: "com.icure.cardinal.sdk/initializers", binaryMessenger: registrar.messenger())
		let filtersChannel = FlutterMethodChannel(name: "com.icure.cardinal.sdk/filters", binaryMessenger: registrar.messenger())
		let paginationChannel = FlutterMethodChannel(name: "com.icure.cardinal.sdk/pagination", binaryMessenger: registrar.messenger())
		let subscriptionChannel = FlutterMethodChannel(name: "com.icure.cardinal.sdk/subscription", binaryMessenger: registrar.messenger())
		let utilsChannel = FlutterMethodChannel(name: "com.icure.cardinal.sdk/utils", binaryMessenger: registrar.messenger())
		apiChannel.setMethodCallHandler(ApisPlugin.handle)
		initializersChannel.setMethodCallHandler(InitializersPlugin.handle)
		paginationChannel.setMethodCallHandler(PaginatedListIteratorPlugin.handle)
		subscriptionChannel.setMethodCallHandler(SubscriptionPlugin.handle)
		utilsChannel.setMethodCallHandler(UtilsPlugin.handle)
		filtersChannel.setMethodCallHandler(FiltersPlugin.handle)
		registrar.addApplicationDelegate(CardinalSdkPlugin())
        let callbackChannel = FlutterMethodChannel(name: "com.icure.cardinal.sdk/dartCallbacks", binaryMessenger: registrar.messenger())
        DartCallbacksHandlerCompanion.shared.registerUsingMethodChannel { method, callbackId, args, resultCallback in
            let arguments: [String: String] = args != nil
              ? ["callbackId": callbackId, "params": args!]
              : ["callbackId": callbackId]
            callbackChannel.invokeMethod(method, arguments: arguments) { dartResult in
                if let stringResult = dartResult as? String {
                    resultCallback(stringResult, nil, nil, nil)
                } else if let dartError = dartResult as? FlutterError {
                    resultCallback(nil, dartError.code, dartError.message, dartError.details != nil ? String(describing: dartError.details!) : nil)
                } else {
                    resultCallback(nil, "UnexpectedDartResult", String(reflecting: dartResult), nil)
                }
            }
        }
	}
	public func detachFromEngine(for registrar: any FlutterPluginRegistrar) {
		ApiScope.shared.teardown()
        DartCallbacksHandlerCompanion.shared.unregister()
	}
}

File: dart-wrapper/cardinal_sdk/android/src/main/kotlin/com/icure/cardinal_sdk/CardinalSdkPlugin.kt
package com.icure.cardinal_sdk

import com.icure.cardinal.sdk.dart.utils.ApiScope
import com.icure.cardinal.sdk.dart.utils.DartCallbacksHandler
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodChannel

class CardinalSdkPlugin: FlutterPlugin {
  private lateinit var apiChannel : MethodChannel
  private lateinit var filtersChannel : MethodChannel
  private lateinit var initializersChannel : MethodChannel
  private lateinit var paginationChannel : MethodChannel
  private lateinit var subscriptionChannel : MethodChannel
  private lateinit var utilsChannel : MethodChannel

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    ApiScope.setup()
    apiChannel = MethodChannel(flutterPluginBinding.binaryMessenger, "com.icure.cardinal.sdk/api")
    apiChannel.setMethodCallHandler(CardinalApisPlugin)
    filtersChannel = MethodChannel(flutterPluginBinding.binaryMessenger, "com.icure.cardinal.sdk/filters")
    filtersChannel.setMethodCallHandler(CardinalFiltersPlugin)
    initializersChannel = MethodChannel(flutterPluginBinding.binaryMessenger, "com.icure.cardinal.sdk/initializers")
    initializersChannel.setMethodCallHandler(InitializersPlugin(flutterPluginBinding.applicationContext))
    paginationChannel = MethodChannel(flutterPluginBinding.binaryMessenger, "com.icure.cardinal.sdk/pagination")
    paginationChannel.setMethodCallHandler(PaginationPlugin)
    subscriptionChannel = MethodChannel(flutterPluginBinding.binaryMessenger, "com.icure.cardinal.sdk/subscription")
    subscriptionChannel.setMethodCallHandler(SubscriptionPlugin)
    utilsChannel = MethodChannel(flutterPluginBinding.binaryMessenger, "com.icure.cardinal.sdk/utils")
    utilsChannel.setMethodCallHandler(UtilsPlugin)
    val callbackMethodChannel = MethodChannel(flutterPluginBinding.binaryMessenger, "com.icure.cardinal.sdk/dartCallbacks")
    DartCallbacksHandler.registerUsingMethodChannel { methodName, callbackId, args, resultCallback ->
      val arguments = listOfNotNull(
        Pair("callbackId", callbackId),
        args?.let { Pair("params", it) }
      ).toMap()
      callbackMethodChannel.invokeMethod(
        methodName,
        arguments,
        object : MethodChannel.Result {
          override fun success(p0: Any?) {
            if (p0 is String) {
              resultCallback(p0, null, null, null)
            } else {
              resultCallback(
                null,
                "UnexpectedSuccessType",
                "Got ${p0?.javaClass?.simpleName}",
                null
              )
            }
          }

          override fun error(p0: String, p1: String?, p2: Any?) {
            resultCallback(null, p0, p1, p2?.toString())
          }

          override fun notImplemented() {
            resultCallback(null, "NotImplemented", null, null)
          }
        }
      )
    }
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    apiChannel.setMethodCallHandler(null)
    initializersChannel.setMethodCallHandler(null)
    paginationChannel.setMethodCallHandler(null)
    subscriptionChannel.setMethodCallHandler(null)
    utilsChannel.setMethodCallHandler(null)
    ApiScope.teardown()
    DartCallbacksHandler.unregister()
  }
}
